lib/dynamoid/transaction/delete.rb

```diff
@@ -0,0 +1,35 @@
+require_relative 'action'
+
+module Dynamoid
+  class Transaction
+    class Delete < Action
+
+      def initialize(model_or_model_class, id = nil, options = {})
+        model = model_or_model_class.is_a?(Dynamoid::Document) ?
+                  model_or_model_class : model_or_model_class.find(id, consistent_read: true)
+        super(model, attributes, options)
+      end
+
+      def run_callbacks
+        model.run_callbacks(:destroy) do
+          yield if block_given?
+        end
```

Comment:
`#delete` in Rails doesn't run callbacks, only `#destroy`

Reply:
Indeed. I will rename this class to destroy and we can also have a delete method which is shorthand for destroy(skip_callbacks: true, skip_validation: true)

Reply:
The main difference (in Rails) between `delete` and `destroy` is that the last one runs callbacks and needs instance. The first one doesn't need to have a model instantiated, but it loads a model now. This seems excessive.